sdcflows/interfaces/bspline.py

```diff
@@ -264,24 +266,28 @@ def _run_interface(self, runtime):
         )
 
         # Generate warp field
-        phaseEncDim = "ijk".index(self.inputs.pe_dir[0])
-        phaseEncSign = [1.0, -1.0][len(self.inputs.pe_dir) != 2]
-
-        data *= phaseEncSign * self.inputs.ro_time
+        pe_axis = "ijk".index(self.inputs.pe_dir[0])
+        pe_sign = -1.0 if self.inputs.pe_dir.endswith("-") else 1.0
+        pe_size = targetnii.header.get_zooms()[pe_axis]
+        data *= pe_sign * self.inputs.ro_time * pe_size
 
         fieldshape = tuple(list(data.shape[:3]) + [3])
         self._results["out_warp"] = fname_presuffix(
             self.inputs.in_target, suffix="_xfm", newpath=runtime.cwd
         )
         # Compose a vector field
         field = np.zeros((data.size, 3), dtype="float32")
-        field[..., phaseEncDim] = data.reshape(-1)
-        aff = targetnii.affine.copy()
-        aff[:3, 3] = 0.0
-        # Multiplying by the affine implicitly applies the voxel size to the shift map
-        field = nb.affines.apply_affine(aff, field).reshape(fieldshape)
+        field[..., pe_axis] = data.reshape(-1)
+
+        # If coordinate system is oblique, project displacements through directions matrix
+        aff = targetnii.affine
+        if (obliquity(aff).min() * 180 / pi) > 0.01:
```

Comment:
Do you want `min` here, not `max`? This implies that it only needs to be plumb on one axis, not all.

Reply:
Good catch, thanks